sky/cli.py

```diff
@@ -3876,6 +3950,459 @@ def spot_dashboard(port: Optional[int]):
             click.echo('Exiting.')
 
 
+@cli.group(cls=_NaturalOrderGroup)
+def serve():
+    """SkyServe commands CLI."""
+    pass
+
+
+@serve.command('up', cls=_DocumentedCodeCommand)
+@click.argument('entrypoint',
+                required=True,
+                type=str,
+                **_get_shell_complete_args(_complete_file_name))
+@click.option('--service-name',
+              '-n',
+              default=None,
+              type=str,
+              help='A service name. Unique for each service. If not provided, '
+              'provision a new service with an autogenerated name.')
+@click.option('--yes',
+              '-y',
+              is_flag=True,
+              default=False,
+              required=False,
+              help='Skip confirmation prompt.')
+def serve_up(
+    entrypoint: str,
+    service_name: Optional[str],
+    yes: bool,
+):
+    """Launch a SkyServe service.
+
+    ENTRYPOINT must point to a valid YAML file.
+
+    Example:
+
+    .. code-block:: bash
+
+        sky serve up service.yaml
+    """
+    if service_name is None:
+        service_name = backend_utils.generate_service_name()
+
+    previous_service_record = global_user_state.get_service_from_name(
+        service_name)
+    if previous_service_record is not None:
+        if previous_service_record['status'] in [
+                status_lib.ServiceStatus.CONTROLLER_FAILED,
+                status_lib.ServiceStatus.FAILED
+        ]:
+            prompt = (f'Service {service_name!r} has failed. '
+                      'Please clean up the service and try again.')
+        else:
+            prompt = f'Service {service_name!r} already exists.'
```

Comment:
Add: Hint: updating a service will be supported in the future. For now, `sky serve down` first and relaunch.